### PR TITLE
Added functionality of choosing between Re/Im and Re in 2D complex plots

### PR DESCRIPTION
**Pull request by FarBo**
This PR suggests the following:

- Two new buttons (Real/Imag and Real) added to have more control for complex 2D plots
- It dynamically switches between Real/Imag, Real and Mag/Phase (selecting one button, disables the other buttons in this category)
- It works consistent with other controls like PlotType, e.g., it is possible to change PlotType together with this new added buttons.

It is tested and working as it should.

The following can be worked later to have more improvements and functionalities related to this topic:
- Imag button can be added with the functionality for showing only imaginary.
- For now, only 2D plots can be used with these new buttons. 1D plots are still showing real and imaginary parts in one figure. This could be improved later.
- The default behavior of 2D plot is Real/Imag, but the Real/Imag button is not activated by default. It is a minor thing, but it's nice to have.

### CODE REVIEW

**Comment by wpfff** [comment]
I like the idea of adding the additional options.
This conflicts with some pending changes on the plotting system in the current form, though :)
I'm proposing to hold off on this and merge this functionality into an updated system (once we determine what exactly that updated system is). @jenshnielsen - thoughts?

**Comment by FarBo** [comment]
@wpfff 
That's right.
As I have heard, the refactoring is going to take some time and in the meantime we have a release (tomorrow if I am not wrong). So, we can have this functionality in this release.

@astafan8

File: plottr/plot/mpl.py
Comment (jenshnielsen) [inline plottr/plot/mpl.py]: Rather than having this state that cannot be selected we should programmatically set the button of the default state to checked

File: plottr/plot/mpl.py
Comment (jenshnielsen) [inline plottr/plot/mpl.py:970]: Please revert back so that this method sets `self.complexPreference`

File: plottr/plot/mpl.py
Comment (jenshnielsen) [inline plottr/plot/mpl.py]: This code does nothing.  The if statements does not compare against anything so they will always evaluate the same way. `is` does not assign to anything that only checks if one object is the same as the other one and return a bool which the code just discards. 

I suggest that you revert back to the original logic

**Review by jenshnielsen — CHANGES_REQUESTED** [review]
> For now, only 2D plots can be used with these new buttons. 1D plots are still showing real and imaginary parts in one figure. This could be improved later.
The default behavior of 2D plot is Real/Imag, but the Real/Imag button is not activated by default. It is a minor thing, but it's nice to have.

IMHO these two are blockers for merging the pr and must be implemented before merging this

File: plottr/plot/mpl.py
Comment (FarBo) [reply plottr/plot/mpl.py:970]: I am going to remove complexPreference completely in the next commit as it is not going to be used anywhere. I am keeping this function as it is, and I fixed the default check. Please look at the next commit.

File: plottr/plot/mpl.py
Comment (FarBo) [reply plottr/plot/mpl.py]: I removed this and included the default activation.

File: plottr/plot/mpl.py
Comment (FarBo) [reply plottr/plot/mpl.py]: I completely removed these conditional statements.

**Comment by FarBo** [comment]
The above commit "2ce47e2" addressed these:
1. Now, we can have complex options for 1D plots both single trace and multitrace
2. The complex options are deactivated when the data is not complex.
3. Some refactoring in 1D plots.

File: plottr/plot/mpl.py
Comment (jenshnielsen) [reply plottr/plot/mpl.py]: Why is empty back now?

Comment (jenshnielsen) [reply plottr/plot/mpl.py]: Please use the relevant selection between the 3 existing (I would guess real and imag ) for 1d plots and disable the 2 other ones rather than adding empty which still has the problems we discussed last week

Comment (FarBo) [reply plottr/plot/mpl.py]: This empty is because of having the complex options deactivated for non-complex data, which is used in line 959.
To handle having the real/imag as the enabled default, I added this function `def setAllowedComplexTypes(self, *args: ComplexRepresentation)` in line 768. So, in that regard we don't have any problem for both 1D and 2D. We also need this function to handle deactivation complex options for non-complex data.

Comment (jenshnielsen) [reply plottr/plot/mpl.py]: I still do not see why this is needed:

* If there is only real data: select real option and gray out the other ones.
* For 1D plot we show real and imag so select real and imag and gray out the other ones.

Comment (FarBo) [reply plottr/plot/mpl.py]: Point 1: That's fine. I also thought about that and tested. 
Pont 2: I am bit confused why we only want real/imag for 1d plot if the data is complex? We also need to have real, mag/phase for 1D plot, right? Am I missing something?

Comment (jenshnielsen) [reply plottr/plot/mpl.py]: Sorry my bad I assumed that you did what we talked about and disabled complex for 1d plots. If you have actually implemented that we can just use `real` for data that is non complex and remove the empty state

**Comment by FarBo** [comment]
@jenshnielsen 

The last commit fixes the logic for complex options, as discussed above. Now:
1. The empty state is removed.
2. If the data is complex: All complex options (Real/Imag, Real, Mag/Phase) are enabled, which Real/Imag is activated by default.
3. If the data is not complex: Only Real is enabled and activated.

File: plottr/plot/mpl.py
Comment (jenshnielsen) [inline plottr/plot/mpl.py]: I suggest that we revert this order now

File: plottr/plot/mpl.py
Comment (jenshnielsen) [inline plottr/plot/mpl.py:671]: I think it would be more logic if the order is. Real, Real/Imag, Mag/Phase rather than having real in the middle

File: plottr/plot/mpl.py
Comment (jenshnielsen) [inline plottr/plot/mpl.py]: Perhaps we can come up with a better name than `args` `complex_representations` perhaps.

**Comment by jenshnielsen** [comment]
Looks good and works well. Left a few minor comments

**Review by jenshnielsen — APPROVED** [review]

**Pull request merged**
